website/content/docs/concepts/oidc-provider.mdx

```diff
@@ -85,39 +88,84 @@ Several named scopes can be made available on an individual provider. Note that
+the differences between confidential and public clients in Vault.
+
+##### Confidential
+
+Confidential clients are capable of maintaining the confidentiality of their credentials.
+Confidential clients have a `client_secret`. The `client_secret` will have a prefix of
+`hvo_secret` followed by 64 random characters in the base62 character set.
+
+Confidential clients may use Proof Key for Code Exchange ([PKCE](https://datatracker.ietf.org/doc/html/rfc7636))
+during the authorization code flow.
+
+Confidential clients must authenticate to the token endpoint using the
+`client_secret_basic` [client authentication method](https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication).
+
+##### Public
+
+Public clients are not capable of maintaining the confidentiality of their credentials.
+As such, public clients do not have a `client_secret`.
+
+Public clients must use Proof Key for Code Exchange ([PKCE](https://datatracker.ietf.org/doc/html/rfc7636))
+during the authorization code flow.
+
+Public clients use the `none` [client authentication method](https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication).
+
 ### Assignments
 
 Assignment resources are referenced by clients via the `assignments` parameter. This parameter limits the set of Vault users allowed to authenticate. The assignments of an associated client are validated during the authentication request, ensuring that the Vault identity associated with the request is a member of the assignment's entities or groups.
 
+Each Vault namespace will contain a built-in assignment resource named `allow_all`. The
+`allow_all` assignment allows all Vault entities to authenticate through a client. The
+`allow_all` assignment cannot be modified or deleted.
+
 ### Keys
 
-Key resources are referenced by clients via the key parameter. This parameter specifies the key that will be used to sign ID tokens for the client. See existing [documentation](/api-docs/secret/identity/tokens#create-a-named-key) for details on keyring management, supported signing algorithms, rotation periods, and verification TTLs. Currently, a key referenced by a client cannot be changed.
+Key resources are referenced by clients via the `key` parameter. This parameter specifies the key that will be used to sign ID tokens for the client. See existing [documentation](/api-docs/secret/identity/tokens#create-a-named-key) for details on keyring management, supported signing algorithms, rotation periods, and verification TTLs. Currently, a key referenced by a client cannot be changed.
+
+Each Vault namespace will contain a built-in key resource named `default`. Clients that don't
+specify the `key` parameter at creation time will use the `default` key. The `default` key
+will use the `RS256` signing algorithm, allow all client IDs, and have rotation and verification
+TTLs of `24h`. The `default` key can be modified but not deleted.
```

Comment:
Wondering if this should be in list form so that it's easier to grok.

Reply:
Good idea. Done in a93251e.